tests/drivers/can/api_additional/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(can_add)
+
+target_include_directories(app PRIVATE ../../../../../zephyr/tests/drivers/can/api/src/)
+
+target_sources(app PRIVATE ../../../../../zephyr/tests/drivers/can/api/src/common.c)
```

Comment:
Maybe you could reference $ENV{ZEPHYR_BASE} or WEST_TOPDIR instead of traversing so many directories up ?